### ISSUE TITLE
Improved explanation for DynamoDB IgnoreNullsMode

### ISSUE THREAD

**Issue by chrylis** [aws/aws-sdk-java-v2#5668]
### Describe the issue

In a recent update, the boolean Update property `ignoreNulls` was deprecated in favor of `ignoreNullsMode`, which takes a strategy. However, I was unable to understand from the documentation what the various options do; it seems to be written from the perspective of someone already familiar with the internals of the framework. In particular, it's not clear to me which values correspond to existing code of `ignoreNulls(true)` or `ignoreNulls(false)`.

Specific items:

1. Mapping instructions for converting existing `ignoreNulls` to `IgnoreNullsMode` values would be helpful, as would a "replaced by" in the Javadoc.
2. In `SCALAR_ONLY`, the phrasing "if done so" is unclear.
3. `SCALAR_ONLY` says that updates to _nested_ attributes are supported but not _maps_; it's unclear whether this means that a nested Java object is supported (since it would be represented as a map in DynamoDB). Does the SDK distinguish between `com.example.MyNestedObject` and `java.util.Map`?
4. `MAPS_ONLY` says that updating scalar values is supported—overwriting existing values in the table. But this behavior doesn't seem to relate to nulls, if the user is providing something. Or is this saying that the scalar nulls will overwrite the stored attributes?
5. `DEFAULT` says it disables "special handling" but not what the default rules are—does `DEFAULT` mean that nulls (for an attempted partial update) on the entity object will cause nulls to overwrite the database record? Or does it mean the opposite?

### Links

https://sdk.amazonaws.com/java/api/2.28.24/software/amazon/awssdk/enhanced/dynamodb/model/IgnoreNullsMode.html

**Comment by tcerda95** [comment]
Hello, can we get a reponse on this? It is unclear to me as well which `IgnoreNullsMode` maps to the deprecated `ignoreNulls(true)` by just reading the documentation

**Comment by Sa1Gur** [comment]
In the mean time better to check the source code. Judging by https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/UpdateItemOperation.java#L101

```
IgnoreNullsMode ignoreNullsMode = request.map(r -> Optional.ofNullable(r.ignoreNullsMode()),
                                                     r -> Optional.ofNullable(r.ignoreNullsMode()))
                                                        .orElse(IgnoreNullsMode.DEFAULT);

        if (ignoreNullsMode == IgnoreNullsMode.SCALAR_ONLY
            || ignoreNullsMode == IgnoreNullsMode.MAPS_ONLY) {
            ignoreNulls = true;
        }
```
false maps to DEFAULT, at least for Update

**Comment by benpeke** [comment]
I agree that the documentation isnt very clear, however there is a more thorough explaination on the user guide: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/ddb-en-client-use.html

**Comment by emakunin** [comment]
Based on the ignored fields table from [the doc](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/ddb-en-client-use.html) The enum and property names seems to be a bit bad and confusing. 
IMHO the properties and enums should be renamed. (or I don't understand the reasoning)

1. `ignoreNullsMode` request param and the Enum should be `UpdateMode`. 
1. Enum values names do not reflect theirs meaning and say nothing about lists. So better to
 `IgnoreNullsMode.DEFAULT` -> `UpdateMode.REMOVE_NULLED`
 `IgnoreNullsMode.SCALAR_ONLY` -> `UpdateMode.IGNORE_NULLED`
 `IgnoreNullsMode.MAPS_ONLY` -> `UpdateMode.REMOVE_NSTED_NULLED_SCALARS`

Type of property | in SCALAR_ONLY mode | in MAPS_ONLY mode | in DEFAULT mode
-- | -- | -- | --
Top scalar | Yes | Yes | No
Bean or map | Yes | Yes | No
Scalar value of a bean or map entry | Yes1 | No2 | No
List or set | Yes | Yes | No

**Comment by JamesMasonNHS** [comment]
I've been working on moving from sdk v1 to v2 for a while and this has caused me no end of headaches, I _think_ I am now aware of what each value does but still not even confident.
A simple way to solve this imo would be to just state what effect each would have on the item as a whole after the transaction somewhere in the description.

 i.e.
DEFAULT-> replaces the item in its entirety, keeping nulls in the provided item.
MAPS_ONLY -> updates only provided attributes of the item at the top level, preserving nulls in bean or map attributes provided.
SCALAR_ONLY -> updates only the provided attributes of the item.

(presuming that is correct, if not something similar that is accurate of course)
